File: backend/app/schemas/project_group.py
from app.schemas import company as schema_company
from pydantic import BaseModel


class ProjectGroup(BaseModel):
    rid: int
    name: str
    detail: str
    company: schema_company.Company

    class Config:
        orm_mode = True


class ProjectGroupCreate(BaseModel):
    rid_companies: int
    name: str
    detail: str

    class Config:
        orm_mode = True


class ProjectGroupUpdate(BaseModel):
    rid: int
    rid_companies: int
    name: str
    detail: str

    class Config:
        orm_mode = True

File: frontend/app/src/types/ProjectGroup.ts
import type { Company } from '@/types/Company'

export interface ProjectGroup {
  rid: number
  company: Company
  name: string
  detail: string
}

export interface ProjectGroupCreate {
  rid_companies: number
  name: string
  detail: string
}

export interface ProjectGroupUpdate {
  rid: number
  rid_companies: number
  name: string
  detail: string
}
